modules/data.remote/R/NASA_DAAC_download.R

```diff
@@ -0,0 +1,110 @@
+#' Parallel download data from the NASA ORNL DAAC server given period, spatial bounding box, and data DOI.
+#'
+#' @param ul_lat Numeric: upper left latitude.
+#' @param lr_lat Numeric: lower right latitude.
+#' @param ul_lon Numeric: upper left longitude.
+#' @param lr_lon Numeric: lower right longitude.
+#' @param ncore Numeric: numbers of core to be used if the maximum core
+#' @param from Character: date from which the data search starts. In the form
+#'   "yyyy-mm-dd".
+#' @param to Character: date on which the data search end. In the form
+#'   "yyyy-mm-dd".
+#' @param outdir Character: path of the directory in which to save the
+#'   downloaded files.Default to the working directory. If it doesn't exist it
+#'   will be created. Ignored if \code{just_path=TRUE}
+#' @param doi Character: data DOI on the NASA DAAC server.
+#' @param netrc_file Character: path to the credential file, default is NULL.
+#' @param just_path Boolean: if we just want the metadata and URL or proceed the actual download.
+#'
+#' @return A list containing meta data for each data downloaded.
+#' @export
+#' 
+#' @examples
+#' @author Dongchen Zhang
+#' @importFrom foreach %dopar%
+NASA_DAAC_download <- function(ul_lat,
+                               ul_lon,
+                               lr_lat,
+                               lr_lon,
+                               ncore = parallel::detectCores() - 1,
```

Comment:
This should default to 1 -- multicore needs the user's explicit request.

Reply:
Fixed.